### ISSUE TITLE
New text editor allows to keep writing code after reaching the code size limit

### ISSUE THREAD

**Issue by nildo** [gpoesia/loopye#75]
In some lessons, there is a code size limit. With the old editor, it was not possible to type more characters than the limit. The new editor allows to type more than the limit. However, when hit the Play button, only the  number of steps up to the limit is executed anyway.

**Comment by gpoesia** [comment]
Talking to @msclar about this, we thought about making the following changes (which would incidentally also fix this issue):

- Allow typing after the limit
- Show the character count in red past the limit (and *not* showing it in red when the limit has been reached)
- When the code is larger than the limit, generate a compilation error

@nildo What do you think?

**Comment by gpoesia** [comment]
Looking at the code, I think this might be a possible not-that-hacky solution:

- Add a `validate(sourceCode) : Array<String>` method in `LessonStepPlayer`'s prototype. For now, this `validate` method will only check for the code size limit, but it might do more in the future. It returns a list of errors found.
- In every implementation of `LessonStepPlayer.play` (in the actual lessons), the first lines should be:

```js
var errors = this.validate(sourceCode);
if (!!errors) {
  return errors;
}
```

If we were using exceptions for doing error handling, a `this.validate(sourceCode)` should be enough. However, since we're returning errors instead, a little bit more of boilerplate is required. I think it's not that bad, since it makes control flow much more predictable still.

@msclar what do you think? Does it make sense, or do you see any other impediments?

**Comment by gpoesia** [comment]
Fixed in #91
